frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -129,25 +142,174 @@ const fieldsForOpenAPI = (openAPI: SwaggerDefinition): OperandField[] => {
           case 'object':
             if (
               _.values(val.properties).some((nestedVal) =>
-                ['object', 'array'].includes(nestedVal.type),
+                _.values(nestedVal.properties).some((thirdLevelVal) =>
```

Comment:
Wondering if you could factor down the `switch` at this point as well, again just to keep it easier to manage.  Its getting pretty big with some significantly lengthy `case` statements, several nested `if` statements, and the nested `returns`.  Naming some funcs will make it easier to review.

Reply:
> Wondering if you could factor down the `switch` at this point as well, again just to keep it easier to manage. Its getting pretty big with some significantly lengthy `case` statements, several nested `if` statements, and the nested `returns`. Naming some funcs will make it easier to review.

@benjaminapetersen Yes, agreed. I have it in my plan to refactor it but wanted to get rid of the typing issues before refactoring. Thanks